pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -61,6 +61,11 @@ func (mon *Monitor) emitCertificateExpirationStatuses(ctx context.Context) error
 		if err != nil {
 			return err
 		}
+
+		if ic.Spec.DefaultCertificate == nil {
+			return fmt.Errorf("ingresscontroller spec invalid, unable to get default certificate name")
```

Comment:
Stupid minor question: would it make sense to clearly state that the default certificate is nil (or empty) in the returned error -same in the test- rather than saying "spec invalid" ? (as we need to be more knowledgeable to know what "invalid" means in this context)

I think nil or empty is clearer and straightforward, and would be less confusing in case the spec had a config that is not valid but the DefaultCertificate is not nil (in case if any).

Also, that statement can fail due to _spec_ being empty (and then _DefaultCertificate_ nil for sure) or _spec_ not being empty and _DefaultCertificate_ being nil. Are we happy with one statement for both cases ? (maybe yes)🙂

Reply:
Happy to change this error message, I didn't put too much thought into it and somewhat-intentionally left it generic while writing the test in case I discovered other scenarios that could cause a nil pointer dereference on the IngressController. I didn't end up seeing any so we could make this message more specific.

Reply:
Nice. Makes sense, thanks!